client/common/cmdline.c

```diff
@@ -853,13 +851,24 @@ static int freerdp_client_command_line_post_filter(void* context,
 	}
 	CommandLineSwitchCase(arg, "smartcard")
 	{
-		char** p;
-		int count;
-		p = freerdp_command_line_parse_comma_separated_values_offset(arg->Value,
-		        &count);
-		p[0] = "smartcard";
-		status = freerdp_client_add_device_channel(settings, count, p);
-		free(p);
+		if (arg->Flags & COMMAND_LINE_VALUE_PRESENT)
+		{
+			char** p;
+			int count;
+			p = freerdp_command_line_parse_comma_separated_values_offset(arg->Value, &count);
+			p[0] = "smartcard";
+			status = freerdp_client_add_device_channel(settings, count, p);
+			free(p);
+		}
+		else
+		{
+			char* p[2];
+			int count;
+			count = 2;
+			p[0] = "smartcard";
+			p[1] = "";
```

Comment:
Then there is something else wrong with argument parsing. When you add the ```:``` it is interpreted as an (empty) argument. The case without should work nonetheless.